CMakeLists.txt

```diff
@@ -1277,7 +1277,7 @@ endif()
 
 # FIXME: Is there any way to get rid of empty_file.c?
 add_executable(       ${ZEPHYR_PREBUILT_EXECUTABLE} misc/empty_file.c)
-target_link_libraries(${ZEPHYR_PREBUILT_EXECUTABLE} ${TOPT} ${PROJECT_BINARY_DIR}/linker.cmd ${PRIV_STACK_LIB} ${zephyr_lnk} ${CODE_RELOCATION_DEP})
+target_link_libraries(${ZEPHYR_PREBUILT_EXECUTABLE} --for-linker ${TOPT} ${PROJECT_BINARY_DIR}/linker.cmd ${PRIV_STACK_LIB} ${zephyr_lnk} ${CODE_RELOCATION_DEP})
```

Comment:
(It's the 1st time I saw this --for-linker option)
* why use that instead of -Wl (which is used in many other places?
* I can see my gcc actually accepts it, even though I cannot find it anywhere in its manual, but I wonder what about older gcc versions out there

Reply:
because clang needs it, as gcc accepts it (tested 6.2 and 8.3) I didn't make it conditional for the sake of readability. Can change it though

Reply:
checked gcc repo, it is part of common options across all languages.

Reply:
Use "Wl," instead, it is supported by clang, or even better ${LINKERFLAGPREFIX}.

It does the same as --for-linker, but with slightly different syntax (CSV).

There is no reason for us to be using more than one syntax for this AFAICT.